### PR TITLE
setup with dropbox, box, and google

### PR DESCRIPTION
**Pull request by jonstorer**

### CODE REVIEW

File: lib/passport-mock/factories/box.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+var utils = require('./utils');
```
Comment (jonstorer) [inline lib/passport-mock/factories/box.js]: not tested

File: lib/passport-mock/factories/dropbox.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+var utils = require('./utils');
```
Comment (jonstorer) [inline lib/passport-mock/factories/dropbox.js]: tested

File: lib/passport-mock/factories/google.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+var utils = require('./utils');
```
Comment (jonstorer) [inline lib/passport-mock/factories/google.js]: not tested

File: lib/passport-mock/factory.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  this._factories = { };
+};
+
+Factory.prototype.add = function (name, fn) {
+  this._factories[name] = fn;
+};
+
+Factory.prototype.build = function () {
+  var args = Array.prototype.slice.call(arguments);
+  return this._factories[args.shift()].apply(null, args);
```
Comment (jonstorer) [inline lib/passport-mock/factory.js]: throw an error if you try to build a factory that doesn't exist

File: lib/passport-mock/strategy.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  this._callbackURL = options.callbackURL,
+  this._factory = new Factory();
+
+  var factories = require('./factories/factories')
+  for (factoryName in factories) {
+    this._factory.add(factoryName, factories[factoryName])
+  }
+}
+
+//util.inherits(MockStrategy, passport.Strategy);
```
Comment (jonstorer) [inline lib/passport-mock/strategy.js]: need to write test to make this true

File: test/passport-mock/factories.test.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+var Factory = require('../../lib/passport-mock/factory')
```
Comment (jonstorer) [inline test/passport-mock/factories.test.js]: rename file to factory.test.js

File: test/passport-mock/factories/dropbox.test.js [hunk 7]
```diff
@@ -0,0 +1,265 @@
+var factory = require('../../../lib/passport-mock/factories/dropbox');
+var expect = require('chai').expect;
+
+describe.only('dropbox profile', function () {
```
Comment (jonstorer) [inline test/passport-mock/factories/dropbox.test.js]: rm only

**Comment by ekump** [comment]
I think this is more code than the actual passport libraries.

**Comment by jonstorer** [comment]
@ekump well it would be

**Pull request closed without merging**